**What was tried:** QA workflow run [#25593511457](https://github.com/svv2014/loop/actions/runs/25593511457) on PR #239 (branch `fix/issue-237-loop`).

**What failed:**
```
No structured test names found
```

<details><summary>Last 20 lines of failure output</summary>

```
qa-merge	Run tests	2026-05-09T05:59:23.0047822Z ok 335 validate: valid minimal workflow passes
qa-merge	Run tests	2026-05-09T05:59:23.1405082Z ok 336 validate: all committed starter workflows pass validation
qa-merge	Run tests	2026-05-09T05:59:23.2011336Z ok 337 validate: missing version field fails
qa-merge	Run tests	2026-05-09T05:59:23.2620818Z ok 338 validate: version=2 fails
qa-merge	Run tests	2026-05-09T05:59:23.3227746Z ok 339 validate: name mismatch with filename fails
qa-merge	Run tests	2026-05-09T05:59:23.3830318Z ok 340 validate: duplicate trigger_label within issue_stages fails
qa-merge	Run tests	2026-05-09T05:59:23.4429776Z ok 341 validate: same trigger_label across issue_stages and pr_stages is allowed
qa-merge	Run tests	2026-05-09T05:59:23.5037200Z ok 342 validate: unknown transition target fails
qa-merge	Run tests	2026-05-09T05:59:23.5635235Z ok 343 validate: transition to terminal label 'done' passes
qa-merge	Run tests	2026-05-09T05:59:23.6253676Z ok 344 validate: decisions approve/reject referencing unknown labels fails
qa-merge	Run tests	2026-05-09T05:59:23.6857001Z ok 345 validate: empty workflow with no stages fails
qa-merge	Run tests	2026-05-09T05:59:23.7464992Z ok 346 validate: stage missing trigger_label fails
qa-merge	Run tests	2026-05-09T05:59:23.8084307Z ok 347 validate: stage missing handler fails
qa-merge	Run tests	2026-05-09T05:59:23.8697016Z ok 348 validate: missing handler script produces warning but passes
qa-merge	Run tests	2026-05-09T05:59:24.0367206Z ok 349 default.yaml: every referenced label is canonical
qa-merge	Run tests	2026-05-09T05:59:24.1080105Z ok 350 minimal.yaml: every referenced label is canonical
qa-merge	Run tests	2026-05-09T05:59:24.1998170Z ok 351 docs-only.yaml: every referenced label is canonical
qa-merge	Run tests	2026-05-09T05:59:24.3436395Z ok 352 all shipped workflows pass loop_workflow_validate
qa-merge	Run tests	2026-05-09T05:59:24.5744038Z ok 353 no deprecated alias appears as a workflow label
qa-merge	Run tests	2026-05-09T05:59:24.6049943Z ##[error]Process completed with exit code 1.
```
</details>

**Options:**
(a) Fix the failing test(s) — update the code or test to make them pass, then push a new commit to this PR.
(b) Skip the test(s) — if the test is flaky or irrelevant to this change, annotate it with the appropriate skip marker and document why.

--- a/scanner/scanner.sh
+++ b/scanner/scanner.sh
@@ -132,7 +132,10 @@ DEDUP_DIR="/tmp/loop-scanner-dedup"
 mkdir -p "$DEDUP_DIR"
 
 _dedup_key() {
-    echo "$1" | md5sum | cut -d' ' -f1
+    printf '%s' "$1" | md5sum 2>/dev/null \
+        || printf '%s' "$1" | md5 -q 2>/dev/null \
+        || printf '%s' "$1" | sha256sum | cut -c1-32
+    true
 }
 
 # emit <json> <dedup_id>

--- a/tests/scanner.bats
+++ b/tests/scanner.bats
@@ -436,6 +436,20 @@ YAML
     [ ! -f "$dispatch_log" ]
 }
 
+@test "emit: second identical event within 30-minute window is suppressed" {
+    local dispatch_log="$BATS_TMPDIR/dispatch-dedup.log"
+    rm -f "$dispatch_log"
+    dispatch_direct() { echo "dispatched" >> "$dispatch_log"; return 0; }
+
+    local json='{"type":"loop.dev_issue","payload":{"slug":"test","repo":"owner/repo"}}'
+    emit "$json" "dev_issue:owner/repo:99"
+    emit "$json" "dev_issue:owner/repo:99"
+
+    local count
+    count=$(wc -l < "$dispatch_log" 2>/dev/null | tr -d ' ')
+    [ "$count" -eq 1 ]
+}
+
 @test "emit: no dedup key file created when dedup_id is empty" {
     local dispatch_log="$BATS_TMPDIR/dispatch3.log"
     rm -f "$dispatch_log"

